projects/igniteui-angular/src/lib/date-range/igx-date-range.component.ts

```diff
@@ -0,0 +1,661 @@
+        if (this.mode === InteractionMode.DropDown) {
+            this.openToggle(this._dropDownOverlaySettings);
+        }
+    }
+
+    /**
+     * Closes the date range picker's dropdown or dialog.
+     *
+     * @example
+     * ```html
+     * <igx-date-range #dateRange [mode]="'dialog'"></igx-date-range>
+     *
+     * <button (click)="dateRange.close()">Close Dialog</button>
+     * ```
+     */
+    public close(): void {
+        if (!this.toggleDirective.collapsed) {
+            this.toggleDirective.close();
+        }
+    }
+
+    /**
+     * Toggles the date range picker's dropdown or dialog
+     *
+     * @example
+     * ```html
+     * <igx-date-range #dateRange [mode]="'dialog'"></igx-date-range>
+     *
+     * <button (click)="dateRange.toggle()">Toggle Dialog</button>
+     * ```
+     */
+    public toggle(): void {
+        if (!this.toggleDirective.collapsed) {
+            this.close();
+        } else {
+            this.open();
+        }
+    }
+
+    /**
+     * The currently selected value / range from the calendar
+     *
+     * @remarks
+     * The current value is of type `DateRange`
+     *
+     * @example
+     * ```typescript
+     * const newValue: DateRange = { start: new Date("2/2/2012"), end: new Date("3/3/2013")};
+     * this.dateRangePicker.value = newValue;
+     * ```
+     */
+    public get value(): DateRange {
+        return this._value;
+    }
+
+    public set value(value: DateRange) {
+        this._value = value;
+        this._onChangeCallback(value);
+        this.updateInputs();
+    }
+
+    /**
+     * Selects a range of dates. If no `endDate` is passed, range is 1 day (only `startDate`)
+     *
+     * @example
+     * ```typescript
+     * public selectFiveDayRange() {
+     *  const today = new Date();
+     *  const inFiveDays = new Date(new Date().setDate(today.getDate() + 5));
+     *  this.dateRange.selectRange(today, inFiveDays);
+     * }
+     * ```
+     */
+    public selectRange(startDate: Date, endDate?: Date): void {
+        endDate = endDate ?? startDate;
+        const dateRange = [startDate, endDate];
+        this.calendar.selectDate(dateRange);
+        this.handleSelection(dateRange);
+    }
+
+    /** @hidden @internal */
+    public writeValue(value: DateRange): void {
+        this.value = value;
+    }
+
+    /** @hidden @internal */
+    public registerOnChange(fn: any): void {
+        this._onChangeCallback = fn;
+    }
+
+    /** @hidden @internal */
+    public registerOnTouched(fn: any): void { }
+
+    /** @hidden */
+    public ngAfterViewInit(): void {
+        switch (this.mode) {
+            case InteractionMode.DropDown:
+                this.attachOnKeydown();
+                this.applyFocusOnClose();
+                this.subscribeToDateEditorEvents();
+                break;
+            case InteractionMode.Dialog:
+                this.applyFocusOnClose();
+                break;
+        }
+        this.subscribeToToggleEvents();
```

Comment:
why do not subscribe to this in the markup like done for `onOpening` and `onClosing`